network/src/peers/peer/peer.rs

```diff
@@ -47,23 +48,47 @@ pub struct Peer {
     pub status: PeerStatus,
     pub quality: PeerQuality,
     pub is_bootnode: bool,
-    #[serde(skip)]
-    pub(super) network_failures: usize,
 }
 
+const FAILURE_EXPIRY_TIME: Duration = Duration::from_secs(15 * 60);
+const FAILURE_THRESHOLD: usize = 5;
+
 impl Peer {
     pub fn new(address: SocketAddr, is_bootnode: bool) -> Self {
         Self {
             address,
             status: PeerStatus::Disconnected,
             quality: Default::default(),
             is_bootnode,
-            network_failures: 0,
         }
     }
 
-    pub fn judge(&self) -> bool {
-        self.quality.rtt_ms > 1500 || self.quality.failures >= 3 || self.quality.is_inactive(chrono::Utc::now())
+    pub fn judge(&mut self) -> bool {
```

Comment:
this name could be more clear, as it returning a `bool` doesn't immediately suggest if this is a positive or negative judgement